File: Sources/windows/yage/capabilities/flip_horizontally_when_going_left.py
from typing import List
from yage.models.capability import Capability
from yage.models.collisions import Collision
from yage.utils.geometry import Vector


class FlipHorizontallyWhenGoingLeft(Capability):
    def do_update(self, collisions: List[Collision], time: float):
        animation, _ = self.subject.animation()
        if animation and animation.facing_direction is not None:
            self._flip_towards(animation.facing_direction)
        else:
            self._flip_towards(self.subject.direction)

    def _flip_towards(self, direction: Vector):
        self.subject.rotation.is_flipped_horizontally = direction.dx < -0.0001

File: Sources/swift/Sources/Yage/Capabilities/FlipHorizontallyWhenGoingLeft.swift
import Combine
import SwiftUI

public class FlipHorizontallyWhenGoingLeft: Capability {
    override public func doUpdate(with collisions: Collisions, after time: TimeInterval) {
        guard let subject else { return }
        flip(for: subject.direction, state: subject.state)
    }

    private func flip(for direction: CGVector, state: EntityState) {
        if case .action(let anim, _) = state, let animDirection = anim.facingDirection {
            flip(for: animDirection)
        } else {
            flip(for: direction)
        }
    }

    private func flip(for direction: CGVector) {
        let isGoingLeft = direction.dx < -0.0001
        subject?.rotation?.isFlippedHorizontally = isGoingLeft
    }
}
